ra/ra_test.go

```diff
@@ -2022,7 +2028,8 @@ func TestRecheckCAAFail(t *testing.T) {
 
 	// NOTE(@cpu): Safe to skip the cast check here because we already checked err
 	// with `berrors.Is(err, berrors.CAA)`
-	berr, _ := err.(*berrors.BoulderError)
+	var berr *berrors.BoulderError
+	errors.As(err, &berr)
```

Comment:
Let's wrap this in an assert:

```suggestion
	test.Assert(errors.As(err, &berr))
```